editor/src/components/canvas/gap-utils.ts

```diff
@@ -197,3 +189,30 @@ export function maybeFlexGapFromElement(
     direction: flexDirection,
   }
 }
+
+export function recurseIntoChildrenOfMapOrFragment(
+  elements: ElementInstanceMetadataMap,
+  target: ElementPath,
+): Array<ElementInstanceMetadata> {
+  let result: Array<ElementInstanceMetadata> = []
+  for (const elementKey in elements) {
+    const element = elements[elementKey]
+    const elementPath = element.elementPath
+    if (EP.isChildOf(elementPath, target) && !EP.isRootElementOfInstance(elementPath)) {
+      const isMapOrFragment = defaultEither(
+        false,
+        mapEither(
+          (e) => e.type === 'JSX_MAP_EXPRESSION' || e.type === 'JSX_FRAGMENT',
```

Comment:
I'd use `treatElementAsFragmentLike` (with sizeless-div-**not**-considered-fragment-like as the flag) instead of special casing _just_ map and fragment, so the same fix can apply to ternaries and in the future more stuff